pkg/proxy/iptables/proxier.go

```diff
@@ -1004,13 +1029,13 @@ func (proxier *Proxier) syncProxyRules() {
 			)
 			if proxier.masqueradeAll {
 				writeLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
-			} else if len(proxier.clusterCIDR) > 0 {
+			} else {
```

Comment:
Should this be `else if proxier.localDetector.IsImplemented()` ?

Reply:
thanks!

Reply:
actually if IsImplemented() is false, the call to JumpIfNotLocal is a no-op. Since there is no other case, I thought we could simplify it to just the direct call. Does it help to retain the IsImplemented check for compatibility on how the code reads?

Reply:
I like it for readability (and you have it in the IPVS equivalent :)